app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,34 @@
+class QueueItemsController < ApplicationController
+  before_action :require_user
+
+  def index
+    @queue_items = current_user.queue_items
+  end
+
+  def create
+    video = Video.find(params[:video_id])
+
+    add_item_to_queue(video) unless item_is_already_queued?(video)
+    redirect_to my_queue_path
+  end
+
+  def destroy
+    queue_item = QueueItem.find(params[:id])
+    remove_item_from_queue(queue_item)
+    redirect_to my_queue_path
+  end
+
+  private
+
+  def add_item_to_queue(video)
+    QueueItem.create!(video_id: params[:video_id], user: current_user, position: current_user.queue_items.count + 1)
+  end
+
+  def item_is_already_queued?(video)
+    !!QueueItem.where("video_id = ? and user_id = ?", params[:video_id], current_user.id).first

```

Comment:
About method item_is_already_queued?, i double bang to get a true or false result although is not really necessary.
```add_item_to_queue(video) unless item_is_already_queued?(video)``` is expecting ```not true``` to add the item to add the item to the queue. The thing is without the double bang you will get either true or nil so it will work but I think it´s more consistent to expect either true or false when using ```?``` as the final character in the name of the method. What do you think about this?